app/views/questionnaire.py

```diff
@@ -491,10 +491,20 @@ def _build_template(current_location, context=None, template=None):
 
     if previous_location is not None and not is_first_block_for_group and not current_location.block_id == 'thank-you':
         previous_url = previous_location.url(metadata)
+    elif current_location.group_id == 'who-lives-here-relationship':
+        previous_url = _relationship_grid_previous_url(current_location.group_instance).url(metadata)
 
     return _render_template(context, current_location.block_id, front_end_navigation, metadata_context, current_location, previous_url, template)
 
 
+def _relationship_grid_previous_url(current_instance):
```

Comment:
I don't think we need to have "_grid" in the method name here, we don't use it elsewhere and there's only 1 type of relationship page. 

It might also make sense to clarify the type of instance that is passed in eg. current_group_instance, given we have other types of instances in the system.